### PR TITLE
chore: return generic errors for aes256cbc encrypt and decrypt

### LINKED ISSUE #499: return generic errors for aes256cbc encrypt and decrypt
**Is your feature request related to a problem? Please describe.**
Generic errors should be returned if there is a problem with encryption or decryption.

**Describe the solution you'd like**
- [ ] Ensure `Encrypt(message mc.PlainContent) (mc.EncryptedContent, error)` returns `cipher.ErrEncrypt` for all errors ref #498 
- [ ] Ensure `Decrypt(data mc.EncryptedContent) (mc.PlainContent, error)` returns `cipher.ErrDecrypt` for all errors ref #498 

**Describe alternatives you've considered**
Returning errors with details in its current state can provide additional information about the failure that should not be returned.

### PR DESCRIPTION
**Pull request by developerfred**
fixes #499

### CODE REVIEW

File: crypto/cipher/aes256cbc/decrypter.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type Decrypter struct {
 func (d Decrypter) Decrypt(data mc.EncryptedContent) (mc.PlainContent, error) {
 	encryptedData, err := bytesDecode(data)
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not convert encryptedData")
+		return nil, mc.ErrDecrypt()
 	}
 
 	return decryptEncryptedData(d.privateKey, encryptedData)
```
Comment (robdefeo) [inline crypto/cipher/aes256cbc/decrypter.go]: `decryptEncryptedData(d.privateKey, encryptedData)` also returns errors they should be `mc.ErrDecrypt`

File: crypto/cipher/aes256cbc/encrypter.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ type Encrypter struct {
 func (e Encrypter) Encrypt(message mc.PlainContent) (mc.EncryptedContent, error) {
```
Comment (robdefeo) [inline crypto/cipher/aes256cbc/encrypter.go]: All errors returned by this function should be `mc.ErrEncrypt`

Comment (developerfred) [reply crypto/cipher/aes256cbc/encrypter.go]: done

File: crypto/cipher/aes256cbc/decrypter.go [hunk 3]
```diff
@@ -40,7 +39,7 @@ type Decrypter struct {
 func (d Decrypter) Decrypt(data mc.EncryptedContent) (mc.PlainContent, error) {
 	encryptedData, err := bytesDecode(data)
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not convert encryptedData")
+		return nil, mc.ErrDecrypt
 	}
 
 	return decryptEncryptedData(d.privateKey, encryptedData)
```
Comment (robdefeo) [inline crypto/cipher/aes256cbc/decrypter.go]: I missed this one, but `decryptEncryptedData(d.privateKey, encryptedData)` also needs to return `mc.ErrDecrypt` as it directly returns an error.

File: crypto/cipher/aes256cbc/decrypter.go [hunk 4]
```diff
@@ -49,42 +48,41 @@ func (d Decrypter) Decrypt(data mc.EncryptedContent) (mc.PlainContent, error) {
...
 	if err != nil {
-		return nil, errors.WithMessage(err, "could not derive shared secret")
+		return nil, mc.ErrDecrypt
 	}
 
 	macKey, encryptionKey := generateMacKeyAndEncryptionKey(sharedSecret)
 	mac, err := generateMac(macKey, data.InitializationVector, *ephemeralPublicKey, data.Ciphertext)
 
 	if err != nil {
-		return nil, errors.WithMessage(err, "generateMac failed")
```
Comment (hasitpbhatt) [inline crypto/cipher/aes256cbc/decrypter.go]: I don't think it's a good idea to mask all errors with the same error. It's okay to return the same error type, but it should not be the same error message for different types of errors, the actual error is getting masked here. It'll be harder to debug the reason just by seeing "error in decryption".

Comment (developerfred) [reply crypto/cipher/aes256cbc/decrypter.go]: I believe that in the future we will create a variable for each type of error, making the code more maintainable.

Comment (robdefeo) [reply crypto/cipher/aes256cbc/decrypter.go]: @hasitpbhatt thank you for taking the time to leave a review on this PR. Sorry for the delay getting back to you I was off for a while. I would love to hear about how you found out about Mailchain and if there is anything I can do to help you. Please feel free to comment here or join our [slack channel](https://github.com/mailchain/mailchain#get-involved)

You raise an excellent point about the outcome from a developer perspective. The reason for masking the error is to reduce the information an attacker might get from an error.  This does need more thought overall so I have opened an issues #532 to discuss this. 

In the meantime @developerfred I will merge this change as it resolves the issue as originally defined. Once #532 is resolved we will come back to this

**Pull request merged**